### PR TITLE
Update multiQC to allow extra configs

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 1]
```diff
@@ -23,8 +23,9 @@ if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input sample
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-ch_multiqc_config   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : file("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
-ch_multiqc_logo     = params.multiqc_logo   ? file( params.multiqc_logo, checkIfExists: true ) : []
+ch_multiqc_config          = file("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
+ch_multiqc_config_custom   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : []
```
Comment (jfy133) [inline nf_core/pipeline-template/workflows/pipeline.nf:27]:
```suggestion
ch_multiqc_custom_config   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : []
```

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 2]
```diff
@@ -102,6 +103,7 @@ workflow {{ short_name|upper }} {
     MULTIQC (
         ch_multiqc_files.collect(),
         ch_multiqc_config,
+        ch_multiqc_config_custom,
```
Comment (jfy133) [inline nf_core/pipeline-template/workflows/pipeline.nf:106]:
```suggestion
        ch_multiqc_custom_config,
```

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 3]
```diff
@@ -23,8 +23,9 @@ if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input sample
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-ch_multiqc_config   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : file("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
-ch_multiqc_logo     = params.multiqc_logo   ? file( params.multiqc_logo, checkIfExists: true ) : []
+ch_multiqc_config          = file("$projectDir/assets/multiqc_config.yml", checkIfExists: true)
+ch_multiqc_custom_config   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : []
```
Comment (jfy133) [inline nf_core/pipeline-template/workflows/pipeline.nf:27]:
```suggestion
ch_multiqc_custom_config   = params.multiqc_config ? file( params.multiqc_config, checkIfExists: true ) : Channel.empty()
```

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 4]
```diff
@@ -23,8 +23,9 @@ if (params.input) { ch_input = file(params.input) } else { exit 1, 'Input sample
...
+ch_multiqc_logo            = params.multiqc_logo   ? file( params.multiqc_logo, checkIfExists: true ) : []
```
Comment (jfy133) [inline nf_core/pipeline-template/workflows/pipeline.nf:28]:
```suggestion
ch_multiqc_logo            = params.multiqc_logo   ? file( params.multiqc_logo, checkIfExists: true ) : Channel.empty()
```

File: nf_core/pipeline-template/modules/nf-core/modules/multiqc/meta.yml [hunk 5]
```diff
@@ -22,6 +22,10 @@ input:
       type: file
       description: Optional config yml for MultiQC
       pattern: "*.{yml,yaml}"
+  - extra_multiqc_config:
+      type: file
+      description: Second optional config yml for MultiQC. Will override common sections in multiqc_config.
```
Comment (ewels) [inline nf_core/pipeline-template/modules/nf-core/modules/multiqc/meta.yml:27]: Argh, this is wrong upstream 🤦🏻

**Review by ewels — APPROVED** [review]
Looks good to me!

Spotted an error in the upstream yaml file description, but no need to hold up this PR for that. Can update later as it won't have any effect on usage in the pipeline script.

**Pull request merged**